### PR TITLE
fix(gatsby-plugin-sharp): Fix defaults handling

### PR DESCRIPTION
**Pull request by ascorbic**
The previous defaults handling didn't work with array values. This replaces the standard lodash `defaultsDeep` with a custom function that replaces rather than merges arrays. Added unit tests to verify.

### CODE REVIEW

File: packages/gatsby-plugin-sharp/README.md [hunk 1]
```diff
@@ -30,18 +30,30 @@ plugins: [
...
-      failOnError: true,
     },
   },
 ]
 ```
 
+## Options
+
+- `defaults`: default values used for `gatsbyImageData` and `StaticImage` from [gatsby-plugin-image](https://www.gatsbyjs.com/plugins/gatsby-plugin-image).
+  Available options are: `formats`,`placeholder`,`quality`,`breakpoints`,`backgroundColor`,`tracedSVGOptions`,`blurredOptions`,`jpgOptions`,`pngOptions`,`webpOptions`,`avifOptions`.
```
Comment (laurieontech) [inline packages/gatsby-plugin-sharp/README.md:51]: layout?

**Review by laurieontech — COMMENTED** [review]
One question. but looks good otherwise!

File: packages/gatsby-plugin-sharp/README.md (hunk 1 above)
Comment (ascorbic) [reply packages/gatsby-plugin-sharp/README.md:51]: I don't think it makes sense as something to be able to change defaults. I left out things that are very specific to individual images: dimensions, aspect ratio and layout

Comment (ascorbic) [reply packages/gatsby-plugin-sharp/README.md:51]: ...and transform options

**Review by laurieontech — APPROVED** [review]

**Comment by ascorbic** [comment]
Published in `gatsby-plugin-sharp@2.14.2`

**Pull request merged**